src/ApiTestCase.php

```diff
@@ -0,0 +1,202 @@
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ * @author Paweł Jędrzęjewski <pawel.jedrzejewski@lakion.com>
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+abstract class ApiTestCase extends WebTestCase
+{
+    /**
+     * @var \Symfony\Bundle\FrameworkBundle\Client
+     */
+    protected $client;
+
+    public function setUp()
+    {
+        $this->client = static::createClient(array(), array('HTTP_Accept' => 'application/json'));
+    }
+
+    public function tearDown()
+    {
+        foreach ($this->client->getContainer()->getMockedServices() as $id => $service) {
+            $this->client->getContainer()->unmock($id);
+        }
+
+        \Mockery::close();
+        $this->client = null;
+
+        parent::tearDown();
+    }
+
+    /**
+     * Gets service from DIC.
+     *
+     * @param string $id
+     *
+     * @return object
+     */
+    protected function get($id)
+    {
+        return $this->client->getContainer()->get($id);
+    }
+
+    /**
+     * @param Response $response
+     * @param int      $statusCode
+     */
+    protected function assertResponseCode(Response $response, $statusCode)
+    {
+        $this->assertEquals($statusCode, $response->getStatusCode());
+    }
+
+    /**
+     * @param Response $response
+     *
+     * @throws \Exception
+     */
+    protected function assertSuccessfulResponse(Response $response)
+    {
+        $openErrorsInBrowser = false;
+        if (isset($_SERVER['OPEN_ERROR_IN_BROWSER'])) {
+            $openErrorsInBrowser = $_SERVER['OPEN_ERROR_IN_BROWSER'];
+        }
+
+        if (in_array($response->getStatusCode(), [400, 401, 403, 404]) && $openErrorsInBrowser) {
+            $filename = rtrim(sys_get_temp_dir(), DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR.uniqid().'.html';
+            file_put_contents($filename, $response->getBody()->getContents());
+            system(sprintf('open %s', escapeshellarg($filename)));
+
+            throw new \Exception('Internal server error.');
+        }
+    }
+
+    /**
+     * @param Response $response
+     * @param int      $statusCode
+     */
+    protected function assertJsonResponse($response, $statusCode = 200)
+    {
+        $this->assertEquals(
+            $statusCode, $response->getStatusCode(),
+            $response->getContent()
+        );
+
+        $this->assertTrue(
+            $response->headers->contains('Content-Type', 'application/json'),
+            $response->headers
+        );
+    }
+
+    /**
+     * Asserts that response has JSON content matching the one given in file.
+     *
+     * @param Response $response
+     * @param string   $filename
+     *
+     * @throws \Exception
+     */
+    protected function assertJsonResponseContent(Response $response, $filename)
+    {
+        if (!isset($_SERVER['EXPECTED_RESPONSE_DIR'])) {
+            throw new \Exception('You have to set a EXPECTED_RESPONSE_DIR variable.');
```

Comment:
How about we try to guess the dir if it is not set? That would be much better DX. We want to standardize the path anyway.